### PR TITLE
Improve CMake configuration

### PR DESCRIPTION
**Pull request by andrea-iob**
I tried to improve and modernize CMake configuration files. Most changes are transparent because only affect how we interact with CMake. However, there are some visible changes:
 - cmake files for find_package are now installed in "share/bitpit/cmake";
 - configuration of installation directories now uses CMAKE_INSTALL_* variables;
 - variable BITPIT_VERBOSE_MAKE is not used anymore (it is possible to use CMAKE_VERBOSE_MAKEFILE directly);
 - selection of static/shared builds is done using the standard BUILD_SHARED_LIBS variable;
 - tests are now enabled/disabled using the CMake variable BUILD_TESTING (the choice of which tests to run  is still done setting BITPIT_ENABLE_INTEGRATION_TESTS and BITPIT_ENABLE_UNIT_TESTS. These variables will only be available when BUILD_TESTING is set to ON);
 - I've bumped the minimum required version to 3.13;
 - compiler warnings are now always displayed.

### CODE REVIEW

**Comment by andrea-iob** [comment]
I forgot to say that now bitpit defines the bitpit::bitpit imported target.  The target allows external projects to use the bitpit library using the following syntax:

find_package(bitpit)
target_link_libraries(external_target PRIVATE bitpit::bitpit)

File: cmake/BITPITConfig.cmake.in [hunk 1]
```diff
@@ -199,3 +202,21 @@ endforeach ()
 # Set programming languages
 #-----------------------------------------------------------------------------
 set(BITPIT_LANGUAGES "@BITPIT_LANGUAGES@")
+
+#-----------------------------------------------------------------------------
+# Create imported target
+#-----------------------------------------------------------------------------
+if(BITPIT_SHARED)
+    set(LIBRARY_TYPE STATIC)
```
Comment (edoardolombardi) [inline cmake/BITPITConfig.cmake.in:210]: if (BITPIT_SHARED)
 set(LIBRARY_TYPE SHARED)  ?

Comment (andrea-iob) [reply cmake/BITPITConfig.cmake.in]: Fixed.

**Review by marcocisternino — APPROVED** [review]

**Comment by marcocisternino** [comment]
Just a remark not directly related to this pull request. When PETSc is found compiled with MPI, cmake is not able to make PETSc pass the tests because MPI is OFF by default. To recover, the user has to turn BITPIT_ENABLE_MPI ON and to turn PETSC_CURRENT OFF before re-configuring.
Usually, PETSc is compiled with MPI support on all HPC clusters, we could think to change BITPIT_ENABLE_MPI default to ON.
It is just a proposal.

**Comment by andrea-iob** [comment]
MPI support is now enabled by default.

**Pull request merged**